modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerMalformedUrlTestCase.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.http.functional.listener;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.is;
+import static org.mule.module.http.api.HttpConstants.HttpStatus;
+import static org.mule.module.http.api.HttpConstants.HttpStatus.BAD_REQUEST;
+import static org.mule.module.http.api.HttpConstants.HttpStatus.OK;
+
+import org.mule.module.http.utils.SocketRequester;
+import org.mule.tck.junit4.FunctionalTestCase;
+import org.mule.tck.junit4.rule.DynamicPort;
+import org.mule.util.IOUtils;
+
+import com.google.common.base.Charsets;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLEncoder;
+
+import org.apache.http.client.fluent.Request;
+import org.apache.http.HttpResponse;
+import org.apache.http.StatusLine;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class HttpListenerMalformedUrlTestCase extends FunctionalTestCase
+{
+
+    public static final String TEST_MALFORMED = "api/ping%";
+    public static final String TEST_ENCODED_SPACE = "test/foo 1 %";
+    public static final String TEST_ENCODED_HASHTAG = "test/foo 1 #";
+    public static final String TEST_ENCODED_PERCENT2 = "test/%24";
+
+    @Rule
+    public DynamicPort httpPort = new DynamicPort("httpPort");
+
+    @Override
+    protected String getConfigFile()
+    {
+        return "http-listener-malformed-url-config.xml";
+    }
+
+    @Test
+    public void returnsBadRequestOnMalformedUrlForWildcardEndpoint() throws Exception
```

Comment:
Add another test validating that if you send a script as part of the rejected path it's encoded in the resulting response body (to validate this is not vulnerable to XSS attacks). Since your request handler is an `ErrorRequestHandler` it should work, but we still need a test in case we forget about that in the future. See `org.mule.module.http.functional.listener.HttpListenerContentTypeTestCase#rejectsAndEscapesInvalidContentTypeWithScript` as a reference.